### ISSUE TITLE
Failed 2SV Authentication at amazon.com

### ISSUE THREAD

**Issue by tgpaul-ha** [custom-components/alexa_media_player#1187]
**Describe the bug**
The steps to authenticate Alexa Media Player with amazon.com results in a failure indicating "There was a problem."

**To Reproduce**
Add Alexa Media Player integration and configure per 2SV authentication as described in the wiki
Follow the instructions on the website.
Note that I am presented with a total of FOUR screens:
- Initial Alexa at amazon.com login (add credentials if not pre-populated from integration configuration; click button to login)
- A second Alexa at amazon.com login with CAPTCHA (add credentials if not pre-populated from previous step; input CAPTCHA; click button to login)
- A screen to request OTP (first screenshot); I only had authenticator app enabled when the screenshot was taken.  I confirmed this issue still exists when a phone text option is available (and the screen correctly reflects these options); click button to request
- Back to the login with error message (second screenshot)

**Expected behavior**
Authentication is successful and Alexa Media Player is able to access the account.

**Screenshots**
See attached

**System details**
 - Home-assistant (version): core-2021.2.1
 - Hassio (Yes/No): Yes HassOS 5.10
 - alexa_media: 3.8.0
 - alexapy (version from `pip show alexapy` or HA startup): Not sure; having trouble finding it
 - Amazon [2FA is enabled]: Y

**Logs**
Attached
I had trouble finding the .html files despite checking "Advanced Debugging" in the integration.  I did not try creating an alexa_media.accounts entry in the config because I thought it might have been redundant with enabling it via the integration screen.  Let me know if this is not the case and whether the HTML in the logs isn't sufficient.

**Additional context**
![alexa1](https://user-images.githubusercontent.com/53988861/107981088-75a9c980-6f8f-11eb-8b42-9776632a8dec.png)
![alexa2](https://user-images.githubusercontent.com/53988861/107981099-780c2380-6f8f-11eb-9e99-ed89cb59a454.png)
![halogs.txt](https://github.com/custom-components/alexa_media_player/files/5985760/halogs.txt)

**Comment by alandtse** [comment]
Please also add a debug entry for `authcaptureproxy` for your logs and make sure you scrub for your username and password since it will display it in your logs.

**Comment by tgpaul-ha** [comment]
Original post updated with requested info.  Link to logs at bottom of post.

**Comment by alandtse** [comment]
Ok thanks. I see the issue. The form action attribute is blank:
```html
<form id="auth-select-device-form" method="post" novalidate action="" class="a-spacing-medium">
```

This means you are directly being sent to Amazon instead of the proxy with the information. Since Amazon has no idea about your login session (because the proxy is the one doing it), it gives you that error.

**Comment by tgpaul-ha** [comment]
OK great, thanks!  So I guess this is a legit bug, or at least something that can be worked on your end?

Also if you get a moment, I'm somewhat interested to understand at a high level what this proxy is doing and how it works (and probably most importantly how/why Amazon doesn't block it, since it seems like they should be able to detect that the request isn't coming from them).

Thanks!

**Comment by alandtse** [comment]
The proxy is a middleman. It connects to Amazon and sends you the exact page it is seeing. When you submit the form, it goes to the proxy, which then passes that info to Amazon. The bug in this case is you are accidentally submitting to Amazon directly but it doesn't know it has a connection with you going so it throws an error.

To Amazon, it looks like the login attempt it coming through your HA server.

**Comment by Giancrafter** [comment]
Also, I can't log in due to me having an amazon.de and not amazon.com Account. The Proxy reports "Invalid Email" to me and I can't get it to log in

**Comment by me1337me** [comment]
using amazon.de and proxy method I dont see the captcha ;(
Also the legacy method doesnt show a captcha. 
Adblock is disabled, different browser used.
Worked before (dont know the version anymore)

![grafik](https://user-images.githubusercontent.com/63295847/108390180-b19f9180-7210-11eb-9676-eb0ba0ff8ccb.png)

**Comment by alandtse** [comment]
Please provide logs for your use cases. I just confirmed that Amazon.de works for the proxy. In fact it's identical to Amazon.com except for the language selection option.

**Comment by me1337me** [comment]
> 
> 
> Please provide logs for your use cases. I just confirmed that Amazon.de works for the proxy. In fact it's identical to Amazon.com except for the language selection option.

Where can I find the logs?

**Comment by tgpaul-ha** [comment]
I successfully upgraded the version (after a few false starts) and Alexa seems to be linked now.

However, there are still some issues/behaviors I wanted to report.
Going through the exact sequence as last time, I still hit the "There was a problem" screen at the end.  I confirmed that the "Enter OTP" screen had the <form> action filled out, but I still hit this error.

I then proceeded to click "sign-in" again from the error screen and was problem with a window to type in the OTP.  I can't recall if I tried that last time.  The pre-populated OTP was expired, so I had to use Authy to capture the QR code too so that I could fill in the right one (I'm using the same code in both HA and Authy).  It came back with all of my devices plus "all devices" and "this device".  I thought maybe the cache screwed something up so I deleted the integration and tried it again.  The same exact sequence happened again, but this time came back with "this device" missing.  I tried a third time and "this device" is still missing.  I'm not sure what that's for, but I thought it might be a useful piece of info.

Great job and I'd love to see this project mature even more.
